src/main/kotlin/org/wa/rceditor/application/viewmodel/MainViewModel.kt

```diff
@@ -1,113 +1,290 @@
 
-    fun removeSource(item: SourceItem) {
-        model.removeSource(item)
-    }
+    private var languageTitle: String by property()
+    val languageTitleProperty = getProperty(MainViewModel::languageTitle)
 
-    fun checkingEntities(): ObservableList<CheckingEntityItem> {
-        return model.checkingEntities()
-    }
+    private var publisher: String by property()
+    val publisherProperty = getProperty(MainViewModel::publisher)
+
+    private var rights: String by property()
+    val rightsProperty = getProperty(MainViewModel::rights)
+
+    private var subject: String by property()
+    val subjectProperty = getProperty(MainViewModel::subject)
+
+    private var title: String by property()
+    val titleProperty = getProperty(MainViewModel::title)
+
+    private var type: String by property()
+    val typeProperty = getProperty(MainViewModel::type)
+
+    private var version: String by property()
+    val versionProperty = getProperty(MainViewModel::version)
+
+    private var checkingLevel: String by property()
+    val checkingLevelProperty = getProperty(MainViewModel::checkingLevel)
+
+    private val contributors = SortedFilteredList<StringProperty>()
+    val contributorsProperty = SimpleListProperty(contributors)
+
+    private val relations = SortedFilteredList<StringProperty>()
+    val relationsProperty = SimpleListProperty(relations)
+
+    private val sources = SortedFilteredList<SourceItem>()
+    val sourcesProperty = SimpleListProperty(sources)
+
+    private val checkingEntities = SortedFilteredList<StringProperty>()
+    val checkingEntitiesProperty = SimpleListProperty(checkingEntities)
+
+    private val projects = SortedFilteredList<ProjectItem>()
+    val projectsProperty = SimpleListProperty(projects)
 
-    fun addCheckingEntity(text: String) {
-        model.addCheckingEntity(text)
+    private var directoryLoaded: Boolean by property()
+    val directoryLoadedProperty = getProperty(MainViewModel::directoryLoaded)
+
+    private var processing: Boolean by property()
+    val processingProperty = getProperty(MainViewModel::processing)
+
+    lateinit var container: ResourceContainer
+
+
+    // ------------- Handlers ---------------- //
+
+    fun handleNewDocumentSelected() {
+        chooseDirectory("Create Resource Container")?.let {
+            processing = true
+            directoryLoaded = false
+            ResourceContainer.create(it)
+                    .observeOn(JavaFxScheduler.platform())
+                    .subscribeOn(Schedulers.io())
+                    .doOnError {
+                        showPopup(DialogFragment.TYPE.ERROR, it.toString())
+                    }
+                    .onErrorComplete()
+                    .doFinally { processing = false }
+                    .subscribe {
+                        container = it
+                        clearViewData()
+                        directoryLoaded = true
+                    }
```

Comment:
I think this is a good change, if I followed our conversation with Joe yesterday. If we look through this function, we see that you are only doing two things. 1) Changing viewModel properties to update the view, and 2) Calling the ResourceContainer create function (effectively a domain level action). You correctly do not have a use case, because the action of `.create` is sufficient for your application and no other app-specific actions were required.